compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor/Records.hs

```diff
@@ -158,13 +158,14 @@ onExp (L o (SectionR _ mid@(isDot -> True) rhs))
     , srcSpanStart o == srcSpanStart (getLoc mid)
     , srcSpanEnd o == srcSpanEnd (getLoc rhs)
     , Just sels <- getSelectors rhs
-    = setL o $ mkParen $ foldl1 (\x y -> noL $ OpApp noE x (mkVar var_dot) y) $ map (mkVar var_getField `mkAppType`) $ reverse sels
+    -- Don't bracket : it came in as a section so it's already enclosed.
```

Comment:
This is not about `:` is it? At least according to the comment above this is about sections on record accessors so `(.foo.bar)`. It is also not clear to me how this ends up looking in the AST. Do we end up with `mkParen (SectionR …)` or does the whole thing parse as `SectionR …`? In the latter case, removing the parenthesis is not going to work.

Reply:
This is about sections. It is lexically impossible to write a "raw" section in Haskell - they must be parenthesized. That means that the expression being transformed is in the context of an enclosing brackets. Bracketing here the causes double ups `((expr))`.

Reply:
To clarify, `(.foo.bar)` is parsed as `HsPar (HsSectionR {...})`.

Reply:
I'll improve the comment.